### PR TITLE
Fix root window size bug on `Android`

### LINKED ISSUE #98088: Change window size at runtime will shrink viewport size in android
### Tested versions

4.3 Stable

### System information

Android

### Issue description

I'm implementing a viewport setting for my game, to let user select and change the viewport size at runtime.

With following project settings of window
image

Change window.size and window.scaling_3d_scale at runtime, will shrink the viewport size, but when I put the game in background and switch back, viewport size will scale to full screen size.

Here is the code I used

```gdscript
        var viewport_width := 960
	var viewport_height := 540
	var scale := 1.0

	match visual:
		HM_VISUAL.VISUAL_LOW:
			viewport_width = 960
			viewport_height = 540
			scale = 1.0
		HM_VISUAL.VISUAL_MED:
			viewport_width = 1280
			viewport_height = 720
			scale = 1.33
		HM_VISUAL.VISUAL_HIGH:
			viewport_width = 1920
			viewport_height = 1080
			scale = 2.0

	var win := get_window()
	win.size = Vector2i(viewport_width, viewport_height)
	win.scaling_3d_scale = scale
```

Here is a video to show the effect

https://github.com/user-attachments/assets/7eaac904-aa4b-4409-8b3d-d739e80e98be

### Steps to reproduce

Like the video shows

1. change the visual to 'low'
2. put the game in background
3. switch back to the game

### Minimal reproduction project (MRP)

[repoissue98088.zip](https://github.com/user-attachments/files/17349086/repoissue98088.zip)

### PR DESCRIPTION
**Pull request by syntaxerror247**
- Fixes #98088

### CODE REVIEW

**Review by m4gr3d — APPROVED** [review]
Looks good! 
Is there a reason the logic is omitting `TOOLS_ENABLED`?

**Comment by syntaxerror247** [comment]
> Is there a reason the logic is omitting `TOOLS_ENABLED`?

No specific reason, I was just unsure about it's affect on Android Editor, that's why i omitted it.

**Review by m4gr3d — APPROVED** [review]

File: scene/main/window.cpp [hunk 1]
```diff
@@ -389,6 +389,12 @@ void Window::move_to_center() {
 }
 
 void Window::set_size(const Size2i &p_size) {
+#if defined(ANDROID_ENABLED)
+	if (!get_parent()) {
+		// Can't set root window size on Android.
+		return;
+	}
+#endif
 	ERR_MAIN_THREAD_GUARD;
```
Comment (akien-mga) [inline scene/main/window.cpp:398]: The thread guard check should still happen before accessing the scene tree with `get_parent`.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by Summersay415** [comment]
This PR caused regression: #100743
I think this check should be moved to `DisplayServer`, or at least changed to (!get_parent && is_inside_tree)

**Comment by Sauermann** [comment]
When this PR is cherrypicked, #100746 should also be cherrypicked.

**Pull request merged**